packages/components/src/utils/a11y/select/select-aria.ts

```diff
@@ -0,0 +1,73 @@
+import { AriaAttributes } from '../../../types';
+
+export const getFilterInputAriaAttributes = (
+  isOpen: boolean,
+  isRequired: boolean,
+  labelId: string,
+  descriptionId: string,
+  dropdownId: string,
+  activeDescendantId?: number
+): AriaAttributes => {
+  return {
+    ...getSelectDropdownButtonAriaAttributes(isOpen, labelId, descriptionId, dropdownId, activeDescendantId),
+    'aria-autocomplete': 'list',
+    ...(isRequired && {
+      'aria-required': 'true',
+    }),
+  };
+};
+
+export const getSelectDropdownButtonAriaAttributes = (
+  isOpen: boolean,
+  labelId: string,
+  descriptionId: string,
+  dropdownId: string,
+  activeDescendantId?: number
+): AriaAttributes => {
+  return {
+    'aria-labelledby': labelId || null,
+    'aria-describedby': descriptionId || null,
+    'aria-haspopup': 'listbox',
+    'aria-expanded': isOpen ? 'true' : 'false',
+    'aria-controls': dropdownId,
+    ...(isOpen &&
+      activeDescendantId !== undefined && {
+        'aria-activedescendant': `option-${activeDescendantId}`,
+      }),
+  };
+};
+
+export const getListAriaAttributes = (
+  label: string,
+  isRequired: boolean,
+  hasFilter: boolean,
+  isOpen: boolean,
+  multiple = false
+): AriaAttributes => {
+  return {
+    role: 'listbox',
+    'aria-label': label,
+    ...(isRequired &&
```

Comment:
Perhaps replace with ternary and `null` fallback instead of object composition to have somewhat consistent style with the function after.